digital_reader.go

```diff
@@ -65,52 +66,23 @@ func (d *DigitalInput) debounceWrapper(p machine.Pin) {
 	d.lastInput = t
 }
 
-// Button is a struct for handling push button behavior.
-type Button struct {
-	Pin           machine.Pin
-	debounceDelay time.Duration
-	lastInput     time.Time
-	callback      func(p machine.Pin)
+type digitalInput struct {
+	DigitalReader
 }
 
-// NewButton creates a new Button struct.
-func NewButton(pin machine.Pin) *Button {
-	pin.Configure(machine.PinConfig{Mode: machine.PinInputPulldown})
-	return &Button{
-		Pin:           pin,
-		lastInput:     time.Now(),
-		debounceDelay: DefaultDebounceDelay,
+func newDigitalInput(pin machine.Pin) *digitalInput {
+	return &digitalInput{
+		newDigitalReader(pin),
 	}
 }
 
-// Handler sets the callback function to be call when the button is pressed.
-func (b *Button) Handler(handler func(p machine.Pin)) {
-	b.HandlerWithDebounce(handler, 0)
-}
-
-// Handler sets the callback function to be call when the button is pressed and debounce delay time has elapsed.
-func (b *Button) HandlerWithDebounce(handler func(p machine.Pin), delay time.Duration) {
-	b.callback = handler
-	b.debounceDelay = delay
-	b.Pin.SetInterrupt(machine.PinFalling, b.debounceWrapper)
+type button struct {
+	DigitalReader
```

Comment:
Unexport this field maybe so that users can't access the field through `EuroPi`. I'd prefer something like this be set on EuroPi initialization, maybe more on that in another comment.